cdap-support-bundle/src/main/java/io/cdap/cdap/support/services/SupportBundleGenerator.java

```diff
@@ -218,4 +307,49 @@ private NamespaceId validNamespace(NamespaceId namespace) throws Exception {
 
     return namespace;
   }
+
+  private SupportBundlePipelineStatus collectSupportBundleTaskStatus(
+    Set<SupportBundleTaskStatus> supportBundleTaskStatusSet) {
+    CollectionState systemLogTaskStatus = CollectionState.INVALID;
+    CollectionState pipelineInfoTaskStatus = CollectionState.INVALID;
+    CollectionState runtimeInfoTaskStatus = CollectionState.INVALID;
+    CollectionState runtimeLogTaskStatus = CollectionState.INVALID;
+    CollectionState vmInfoTaskStatus = CollectionState.INVALID;
+    Map<SupportBundleTaskType, CollectionState> taskStatusMap = new HashMap<>();
```

Comment:
What's the purpose of putting the task statuses into a map first? Can we just directly assign them to our relevant statuses, i.e. (assuming we don't need to worry about subtasks):
```
for (SupportBundleTaskStatus supportBundleTaskStatus : supportBundleTaskStatusSet) {
  SupportBundleTaskType taskType = SupportBundleTaskType.valueOf(supportBundleTaskStatus.getType());
  switch (taskType) {
        case SupportBundleSystemLogTask:
          systemLogTaskStatus = supportBundleTaskStatus.getStatus();
          break;
        case SupportBundlePipelineInfoTask:
          pipelineInfoTaskStatus = supportBundleTaskStatus.getStatus();
          break;
        case SupportBundleRuntimeInfoTask:
          runtimeInfoTaskStatus = supportBundleTaskStatus.getStatus();
          break;
        case SupportBundlePipelineRunLogTask:
          runtimeLogTaskStatus = supportBundleTaskStatus.getStatus();
          break;
        case SupportBundleVMInfoTask:
          vmInfoTaskStatus = supportBundleTaskStatus.getStatus();
          break;
      }
      SupportBundlePipelineStatus supportBundlePipelineStatus =
      new SupportBundlePipelineStatus(systemLogTaskStatus, pipelineInfoTaskStatus, runtimeInfoTaskStatus,
                                      runtimeLogTaskStatus, vmInfoTaskStatus);
      return supportBundlePipelineStatus;
}
```

Alternatively, if you do want to pass all subtasks back in the status response object, I'd recommend changing the signature of `SupportBundlePipelineStatus` to support directly passing a map, and then add a `parent` field to the `SupportBundleTaskStatus` class which identifies which task is a given subtask's parent.

Reply:
Because we don't know which task is which inside the SupportBundleTaskStatus, so we need a switch to identify the task and put them in correct order for the SupportBundlePipelineStatus constructor

Reply:
I will remove the map

Reply:
Bump